cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceConfigMapUtilsTest.java

```diff
@@ -0,0 +1,174 @@
+import static io.strimzi.operator.cluster.operator.assembly.KafkaRebalanceConfigMapUtils.EXECUTOR_STATE_KEY;
+import static io.strimzi.operator.cluster.operator.assembly.KafkaRebalanceConfigMapUtils.TIME_COMPLETED;
+import static io.strimzi.operator.cluster.operator.assembly.KafkaRebalanceConfigMapUtils.updateRebalanceConfigMapWithProgressFields;
+import static io.strimzi.operator.cluster.operator.assembly.KafkaRebalanceStatusTest.buildOptimizationProposal;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class KafkaRebalanceConfigMapUtilsTest {
+
+    private static final String BROKER_LOAD = buildOptimizationProposal().toString();
+    private static final Map<String, String> BROKER_LOAD_MAP = Map.of(BROKER_LOAD_KEY, buildOptimizationProposal().toString());
+
+    private static ObjectNode createExecutorState(Map<String, String> executorState) throws Exception {
+        ObjectMapper objectMapper = new ObjectMapper();
+        ObjectNode objectNode = objectMapper.createObjectNode();
+        if (executorState != null) {
+            for (Map.Entry<String, String> entry : executorState.entrySet()) {
+                objectNode.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return objectNode;
+    }
+
+    private static ConfigMap createKafkaRebalanceConfigMap(Map<String, String> brokerLoad) {
+        return new ConfigMapBuilder()
+                .addToData(brokerLoad)
+                .build();
+    }
+
+    private static String triggeredTaskTime(int secondsAgo) {
+        return Instant.ofEpochSecond(Instant.now().getEpochSecond() - secondsAgo).toString();
+    }
+
+    @Test
+    public void testProgressFieldsForProposalReadyState() throws Exception {
+        JsonNode es = createExecutorState(null);
+        ConfigMap cm = createKafkaRebalanceConfigMap(BROKER_LOAD_MAP);
+        updateRebalanceConfigMapWithProgressFields(KafkaRebalanceState.ProposalReady, null, es, cm);
+
+        Map<String, String> m =  cm.getData();
+        assertThat(m.containsKey(ESTIMATED_TIME_TO_COMPLETION_KEY), is(false));
+        assertThat(m.get(COMPLETED_BYTE_MOVEMENT_KEY), is("0"));
+        assertThat(m.containsKey(EXECUTOR_STATE_KEY), is(false));
+        assertThat(m.get(BROKER_LOAD_KEY), is(BROKER_LOAD));
+    }
+
+    @Test
+    public void testProgressFieldsForProposalRebalancingState() throws Exception {
+        ZonedDateTime taskStartTime = ZonedDateTime.now().minusSeconds(2);
+        JsonNode es0 = createExecutorState(Map.of("finishedDataMovement",  "250",
+                "totalDataToMove", "10000"));
+        ConfigMap cm = createKafkaRebalanceConfigMap(BROKER_LOAD_MAP);
+        updateRebalanceConfigMapWithProgressFields(KafkaRebalanceState.Rebalancing, taskStartTime, es0, cm);
+
+        Map<String, String> m =  cm.getData();
+        assertThat(m.get(ESTIMATED_TIME_TO_COMPLETION_KEY), is("1"));
```

Comment:
So I ran the numbers myself and checked that the actual time to completion is 78 seconds (approx 1.3 mins). Which explains why this is floored down to "1" min. But it would be nice to put a comment/javadoc for each tests which lists the values of the various parameters and what the expected outcome is. 
```java
/**
 * total data to move: 10000 MB
 * time since task start: 2 secs
 * data moved: 250 MB
 * data rate: 125 MB/s
 * data to move: 9750 MB
 * expected time to completion: 78 seconds
 */
```
This will make it easier for others to parse what the test is doing.

Reply:
Fixed.